lib/bugsnag/loggers/bugsnag_logger.rb

```diff
@@ -0,0 +1,124 @@
+require "bugsnag"
+require "bugsnag/breadcrumbs/breadcrumb"
+
+module Bugsnag::Loggers
+
+  LEVELS = [
+    "debug",
+    "info",
+    "warn",
+    "error",
+    "fatal"
+  ]
+
+  class BugsnagLogger
+
+    attr_reader :level
+
+    def initialize(level="info")
```

Comment:
We havent inherited from logger which is potentially a better idea. I looked at some things. Found log4r which might be a good thing to use? We could also instead build an appender? http://www.rubydoc.info/gems/logging/Logging/Appenders 

That way we dont have to even worry about multi loggers. I havent looked into this deeply at all however!

Reply:
I'll subclass Appender/IO for the same goal in a different branch.  If we'd prefer that I'll close this one out

Reply:
Hmm, I didn't notice when I last checked that Appenders doesn't belong to Stdlib.  Personally I'd prefer to implement an stdlib solution with examples for other common third parties.  I can remove multi-logger and re-tool Bugsnag-Logger into just Logger.

Reply:
There's two sets of users here that we can support:

* folks using the stdlib logger
* folks using other loggers (e.g. logging and log4r gems)

For the first group, the simplest solution seems to be subclassing [`Logger`](http://ruby-doc.org/stdlib-2.4.2/libdoc/logger/rdoc/Logger.html), since it already provides log levels and filtering by level. In that case the implementation only needs to map levels to severities and make breacrumbs/reports from the `log()` method when its invoked.

For the folks using something other than the stdlib Logger, the [logging](https://github.com/TwP/logging) gem seems to be the most used and maintained, as log4r hasn't seen a release in the last 7ish years. We'd need an appender for this and a check that the `Logging` class exists to activate it. The appender and the stdlib Logger subclass could share the same logic, potentially, depending on what an appender needs to do.

Reply:
stdlib includes a [`LogDevice`](https://ruby-doc.org/stdlib-2.4.0/libdoc/logger/rdoc/Logger/LogDevice.html) which seems to be equivalent to an Appender in the Logging lib.  I think targeting these two and allow users to configure the actual logger may be the best solution.